src/MongoDB.Driver.Core/Core/Authentication/ISecurityContext.cs

```diff
@@ -0,0 +1,25 @@
+/* Copyright 2010-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+namespace MongoDB.Driver.Core.Authentication
+{
+    internal interface ISecurityContext
+    {
+        byte[] Next(byte[] challenge);
+        bool IsInitialized { get; }
+        byte[] DecryptMessage(int messageLength, byte[] encryptedBytes);
```

Comment:
Seems like `0` is always passed as the messageLength.  Can we just remove the parameter entirely?

Reply:
Valid question. Analyzing `SspiSecurityContext.DecryptMessage`, the incoming `encryptedBytes` are split between an encrypted message and security trailer. `messageLength` indicates where the break point between the two sections in the byte array. This looks like peculiarities of the Win32 SSPI calls leaking out into the interface.

With `messageLength` being zero, all we are doing is verifying that the received security trailer can be successfully decrypted. As well the returned `decryptedBytes` are actually never used - which should be empty anyways because the incoming encrypted message had zero length.

This area of the code is a good candidate for additional refactoring, which I will pursue while integrating libgssapi.